src/shared-api-objects/current-user.ts

```diff
@@ -20,13 +21,31 @@ function getCurrentUser(
 ): Observable<LoggedInUser | UnauthenticatedUser> {
   return userSubject.asObservable().pipe(
     mergeAll(),
+    tap(setUserLanguage),
     map((r: LoggedInUserFetchResponse) =>
       opts.includeAuthStatus ? r.data : r.data.user
     ),
     filter(Boolean)
   ) as Observable<LoggedInUser | UnauthenticatedUser>;
 }
 
+function setUserLanguage(sessionResponse) {
+  if (sessionResponse?.data?.user?.userProperties?.defaultLocale) {
+    const locale = sessionResponse.data.user.userProperties.defaultLocale;
+    const htmlLang = document.documentElement.getAttribute("lang");
+    if (locale != htmlLang) {
+      document.documentElement.setAttribute("lang", locale);
+      //@ts-ignore
+      (i18n.default || i18n).changeLanguage().catch(e => {
```

Comment:
Ok so no mutation observer?

Should we have the whole language logic in one module and here only the adjustment of the attribute?

Reply:
Aw crap, I forgot that detail :laughing:  I'll set up the mutation observer.

Reply:
https://github.com/openmrs/openmrs-esm-root-config/pull/38